gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -238,10 +251,31 @@ private Completable handlePhotoCaptureResult(String fieldId) {
         .ignoreElements();
   }
 
-  private Observable<File> saveBitmapAndUpdateResponse(Observable<Bitmap> source, String fieldId) {
+  private Observable<String> saveBitmapAndUpdateResponse(
+      Observable<Bitmap> source, String fieldId) {
     return source
         .map(bitmap -> fileUtil.saveBitmap(bitmap, fieldId + ".jpg"))
-        .doOnNext(file -> onTextChanged(form.getValue().getField(fieldId).get(), file.getPath()));
+        .map(
+            file -> {
+              // If offline, Firebase will automatically upload the image when the network
+              // connectivity is  re-established.
+              String remotePath = getRemoteImageDir() + file.getName();
+              return firestoreStorageManager.uploadMediaFromFile(file, remotePath);
```

Comment:
Do I understand correctly that we're currently relying on Firebase to manage offline support?

If so, can we add a TODO and a bug to implement our own offline photos functionality using Android Workers and the local db?